src/movies.js

```diff
@@ -1,16 +1,95 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors (movies){
+    let directors = movies.map(function(movie){
+        return movies.directors;
+    });
+    return directors;
+}   
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies){
+    let dramaMovies = movies.filter(function(movie){
+        movie.director == "Steven Spielberg" && movie.genre.includes("Drama");
+    });
+    return dramaMovies.length;
+}
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movie){
```

Comment:
You are using movies during the function, however the parameter in this cased was named movie. You can note that the parameter is never being used because in your VS code it should look gray.